web-ui/src/main/resources/catalog/components/search/formfields/FormFieldsDirective.js

```diff
@@ -411,10 +412,19 @@
             values: "=gnSortbyValues"
           },
           link: function (scope, element, attrs, searchFormCtrl) {
+            scope.lang = attrs["lang"] || gnGlobalSettings.gnCfg.langDetector.default;
             scope.sortBy = function (v) {
               angular.extend(scope.params, v);
               searchFormCtrl.triggerSearch(true);
             };
+            // Replace the placeholder for the language in the sort field name
+            for (var i = 0; i < scope.values.length; i++) {
+              scope.values[i].sortBy = scope.values[i].sortBy.replace(
```

Comment:
Maybe better to replace it when we build the search query
https://github.com/GeoCat/core-geonetwork/blob/44-sortbyfields-language-placeholder/web-ui/src/main/resources/catalog/components/elasticsearch/EsService.js#L196
because the search is not always using the UI language depending on the configuration https://github.com/GeoCat/core-geonetwork/blob/44-sortbyfields-language-placeholder/web-ui/src/main/resources/catalog/components/elasticsearch/EsService.js#L47-L93